lib/FunctionIsolation/EnforceABI.cpp

```diff
@@ -834,9 +918,58 @@ void EnforceABIImpl::generateCall(IRBuilder<> &Builder,
         ReturnCSVs.push_back(CSV);
       }
     }
+
+    BasicBlock *BB = Builder.GetInsertBlock();
+    Function *Caller = BB->getParent();
+
+    // If we are in a basic block of an isolated function,
+    // we generate an invoke instruction. Do not emit an invoke instruction
+    // if DisableSafetyChecks is on.
+    // TODO: Review second condition
+    if (!DisableSafetyChecks && Caller->getName().startswith("bb.")
+          && !BB->getName().endswith("__continuation")) {
+      BasicBlock *NormalEdge;
+      std::string NormalEdgeName = std::string(BB->getName()) + "__continuation";
```

Comment:
Use a `Twine`

Reply:
Note that `NormalEdgeName` is used twice (there may be two different `NormalEdge` basic blocks depending on if it's part of the `function_dispatcher` or not), so I would need to create two temporary objects here, like `BasicBlock::Create(Context, Twine(BB->getName()) + "__continuation", ...)` and do the same with `BB->splitBasicBlock()`, looks bad style to me (Twine is meant to be used only for passing strings as const refs in parameters). So, either I use a `const std::string &Name = ` or I create a tmp like `(Twine(BB->getName()) + "__continuation").str();`. Not sure if the first allows to avoid a temporary allocation.

Reply:
Just to conclude that, after experimenting a bit, it turned out that:
• `const std::string Name = BB.getName() + "__cont";` doesn't compile (no implicit conversion from `llvm::Twine` to `std::string`);
• `const auto Name = BB.getName() + "__cont";` works but it is not safe at all (the outer Twine refers to the inner temporary Twine object that has already been destroyed, so dangling pointer);
• `auto Name = (Twine(BB->getName()) + "__cont").str();` works, the temporary object is materialized into a string and is literally the same as the original construction (line 932).

Reply:
Gotcha, makes sense to use a `string` then!